e2e-tests/spring-examples/src/test/java/org/evomaster/e2etests/spring/examples/sort/SortEMTest.java

```diff
@@ -0,0 +1,67 @@
+package org.evomaster.e2etests.spring.examples.sort;
+
+import org.evomaster.core.Main;
+import org.evomaster.core.output.TestCase;
+import org.evomaster.core.output.TestSuiteOrganizer;
+import org.evomaster.core.problem.rest.HttpVerb;
+import org.evomaster.core.problem.rest.RestIndividual;
+import org.evomaster.core.search.Solution;
+import org.evomaster.e2etests.spring.examples.namedresource.NRTestBase;
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Note: This class uses the NRTestBase and NamedResourceRest SUT. This is more of a shortcut, since that system seems to
+ * have the diversity of test cases needed.
+ */
+
+public class SortEMTest extends NRTestBase {
+
+    @Test
+    public void testRunEM() throws Throwable {
+
+        runTestHandlingFlakyAndCompilation(
+                "SortEM",
+                "org.bar.SortEM",
+                3_000,
+                (args) -> {
+
+                    Solution<RestIndividual> solution = initAndRun(args);
+
+                    assertTrue(solution.getIndividuals().size() >= 1);
+
+                    TestSuiteOrganizer organizer = new TestSuiteOrganizer();
+
+                    List<TestCase> tclist = organizer.sortTests(solution, true);
+
+                    Iterator<TestCase> iterator = tclist.iterator();
+                    TestCase current, previous = iterator.next();
+                    while(iterator.hasNext()){
+                        current = iterator.next();
+                        // Check that a TC with 500 in the name does not follow a TC without a 500 in the name (500 should be first).
+                        if(current.getName().contains("500")){
+                            assertTrue(previous.getName().contains("500"));
+                        }
+                        // Check that a TC with 400 in the name is preceded by another 400 or by a 500.
+                        if(current.getName().contains("400")){
```

Comment:
by default, 4xx is less important than 2xx codes. 2xx should come before the 4xx in the sorted tests

Reply:
Yes. The code was right, but the test code was wrong.

Reply:
but if the test was wrong, why didn't it fail?

Reply:
I was daft. The test code only looked at the name of the test cases (and, naturally, most info is not there). 

I am now fixing both the test and the sorting (since the updated test helped me find a bug in the sorting part).